lang/cs/Org.Apache.REEF.IO/FileSystem/AzureDataLake/AzureDataLakeFileSystem.cs

```diff
@@ -123,10 +130,17 @@ public void CopyToLocal(Uri remoteFileUri, string localName)
         /// <exception cref="AdlsException">If copy process encounters any exception</exception>
         public void CopyFromLocal(string localFileName, Uri remoteFileUri)
         {
-            TransferStatus status = _adlsClient.BulkUpload(localFileName, remoteFileUri.AbsolutePath);
-            if (status.EntriesFailed.Count != 0)
+            try
+            {
+                TransferStatus status = _adlsClient.BulkUpload(localFileName, remoteFileUri.AbsolutePath);
```

Comment:
I don't think this is synchronous. We need to make it so.

Reply:
Just tested this with a 250Mb file. execution waits on the BulkUpload till all work is done